src/Sylius/Bundle/CoreBundle/Fixture/Factory/PromotionExampleFactory.php

```diff
@@ -142,6 +153,41 @@ protected function configureOptions(OptionsResolver $resolver): void
 
                 return $actions;
             })
+
+            ->setDefault('coupons', [])
+            ->setAllowedTypes('coupons', 'array')
+            ->setNormalizer('coupons', self::getCouponNormalizer($this->couponFactory))
         ;
     }
+
+    /**
+     * @param FactoryInterface $couponFactory
+     *
+     * @return \Closure
+     */
+    private static function getCouponNormalizer (FactoryInterface $couponFactory): \Closure {
```

Comment:
```suggestion
    private static function getCouponNormalizer (?FactoryInterface $couponFactory): \Closure {
```

And we need to provide an exception if this function is used and `$this->couponFactory` is null